### PR TITLE
Show warning for old entity spec version

### PR DESCRIPTION
**Pull request by ktuite**
Frontend of https://github.com/getodk/central-backend/pull/1272 

Issue https://github.com/getodk/central/issues/730

Screenshot 2024-11-19 at 4 31 23 PM

#### Before submitting this PR, please make sure you have:

- [ ] run `npm run test` and `npm run lint` and confirmed all checks still pass OR confirm CircleCI build passes
- [ ] verified that any code or assets from external sources are properly credited in comments or that everything is internally sourced

### CODE REVIEW

File: src/components/form/new.vue [hunk 1]
```diff
@@ -318,7 +330,8 @@ export default {
       "Workflow warnings:",
       {
         "deletedFormExists": "There is a form with ID \"{value}\" in the Trash. If you upload this Form, you won’t be able to undelete the other one with the matching ID.",
-        "structureChanged": "The following fields have been deleted, renamed or are now in different groups or repeats. These fields will not be visible in the Submission table or included in exports by default."
+        "structureChanged": "The following fields have been deleted, renamed or are now in different groups or repeats. These fields will not be visible in the Submission table or included in exports by default.",
+        "oldEntityVersion": "Entities specification version \"{value}\" is not compatible with Offline Entities. Please use version 2024.1.0 or later."
```
[the remark this answers is not in the document]
Comment (ktuite) [reply src/components/form/new.vue]: Should I change it for the `deletedFormExists` string above, too, or would that require it to get re-translated (so maybe I should just leave it as is)?

File: src/components/form/new.vue [hunk 2]
```diff
@@ -41,8 +41,13 @@ definition for an existing form -->
...
             <li v-for="warning of warnings.workflowWarnings">
-              {{ $t('warningsText[3].' + warning.type, { value: warning.details.xmlFormId }) }}
-              <doc-link :to="documentLinks[warning.type]">{{ $t('moreInfo.learnMore') }}</doc-link>
+              {{ $t('warningsText[3].' + warning.type, { value: getWarningValue(warning) }) }}
+              <template v-if="warning.type === 'oldEntityVersion'">
+                <a :href="externalLinks.oldEntityVersion" target="_blank">{{ $t('moreInfo.learnMore') }}</a>
+              </template>
+              <template v-else>
+                <doc-link :to="documentLinks[warning.type]">{{ $t('moreInfo.learnMore') }}</doc-link>
+              </template>
```
[the remark this answers is not in the document]
Comment (ktuite) [reply src/components/form/new.vue]: I like this a lot better, I was struggling with how to manage these different, yet in some ways similar, cases.

File: src/components/form/new.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (matthew-white) [reply src/components/form/new.vue]: Yeah, I think it's not worth it getting re-translated. It'll be nice to have curly quotes in the new warning though.

File: src/components/form/new.vue [hunk 3]
```diff
@@ -41,12 +41,22 @@ definition for an existing form -->
...
+              <template v-else-if="warning.type === 'structureChanged'">
+                {{ $t('warningsText[3].structureChanged') }}
+                <doc-link to="central-forms/#central-forms-updates">{{ $t('moreInfo.learnMore') }}</doc-link>
+                <span>
+                  <br>
+                  <strong>{{ $t('fields') }}</strong> {{ warning.details.join(', ') }}
+                </span>
+              </template>
+              <template v-else-if="warning.type === 'oldEntityVersion'">
+                {{ $t('warningsText[3].oldEntityVersion', { value: warning.details.version }) }}
```
Comment (ktuite) [inline src/components/form/new.vue:57]: could call `value` `version` in the translation message

**Review by matthew-white — APPROVED** [review]

File: src/components/form/new.vue [hunk 4]
```diff
@@ -318,7 +324,8 @@ export default {
       "Workflow warnings:",
       {
         "deletedFormExists": "There is a form with ID \"{value}\" in the Trash. If you upload this Form, you won’t be able to undelete the other one with the matching ID.",
-        "structureChanged": "The following fields have been deleted, renamed or are now in different groups or repeats. These fields will not be visible in the Submission table or included in exports by default."
+        "structureChanged": "The following fields have been deleted, renamed or are now in different groups or repeats. These fields will not be visible in the Submission table or included in exports by default.",
+        "oldEntityVersion": "Entities specification version “{value}” is not compatible with Offline Entities. Please use version 2024.1.0 or later."
```
Comment (ktuite) [inline src/components/form/new.vue:328]: Could possibly say "We recommend using version 2024.1.0 or later..." to leave it more open ended?

**Pull request merged**